### DISCUSSION TITLE
Preset/custom resolutions in windowed mode (like stable)

### DISCUSSION THREAD

**Discussion by jvyden** [ppy/osu#13885] [Ideas]
Hi,

I strongly prefer to play in a smaller 4:3 window when playing on stable, but in lazer there's currently no option for resolutions in windowed mode like in stable.

My current workaround is to use the screen scaling option for the whole window while maximized, but it's pretty hard to get right and it's prone to changing considering the window is free-form and the scaling is a percentage.

To clarify, I don't think the free-form on window should be *removed*, it would definitely be best if the options co-existed somehow, such as an option to lock/snap the window into a given size.

Peppy brought up some concerns with and that adding this option would make it seem as if it were fully supported which is not the case due to latency issues, but in my opinion it should be okay if a warning about the latency [like in stable](https://i.imgur.com/6p4RsVu.png) is shown to the user.

**Comment by frenzibyte** [comment]
There was an attempt at this here https://github.com/ppy/osu/pull/9139 but the PR was closed due to a seemingly bad force-push, and doesn't look to have been submitted again.

**Reply by jvyden** [reply]
I believe I found a branch with their changes [here](https://github.com/jubiman/osu/tree/WindowDropdownMenu). Looks to be pretty far behind, though. (12656 commits behind)

I guess they were trying to override their master with ppy:master and switch the merge request to a new branch but failed.

**Comment by frenzibyte** [comment]
Tracking in https://github.com/ppy/osu/issues/17999
